opsdroid/connector/twitch/__init__.py

```diff
@@ -0,0 +1,576 @@
+
+import opsdroid.connector.twitch.events as twitch_event
+
+from opsdroid.connector import Connector, register_event
+from opsdroid.events import Message
+from opsdroid.const import TWITCH_API_ENDPOINT, TWITCH_WEBHOOK_ENDPOINT, DEFAULT_ROOT_PATH
+
+
+CONFIG_SCHEMA = {
+    Required("code"): str,
+    Required("client-id"): str,
+    Required("client-secret"): str,
+    Required("channel"): str,
+    "redirect": str,
+    "forward-url": str
+}
+
+
+_LOGGER = logging.getLogger(__name__)
+TWITCH_JSON = os.path.join(DEFAULT_ROOT_PATH, 'twitch.json')
+
+
+def build_api_url(client_id, client_secret, authorization, redirect):
+    """Build api url to authenticate and refresh token."""
+    clientid = f"?client_id={client_id}"
+    secret = f"&client_secret={client_secret}"
+    grant = f"&grant_type={authorization}"
+    redirect = f"&redirect_uri={redirect}"
+
+    return f"{TWITCH_API_ENDPOINT}{clientid}{secret}{grant}{redirect}"
+
+async def get_user_id(channel, token, client_id):
+    """Calls twitch api to get broadcaster user id.
+    
+    A lot of webhooks expect you to pass your user id in order to get the 
+    notification when a user subscribes or folllows the broadcaster
+    channel.
+    
+    """
+    url = f"https://api.twitch.tv/helix/users?login={channel}"
+    async with aiohttp.ClientSession() as session:
+        response = await session.get(url, headers={
+            'Authorization': f"Bearer {token}",
+            'Client-ID': client_id
+        })
+        
+        if response.status != 200:
+            _LOGGER.warning(_("Unable to receive broadcaster id - Error: %s, %s."), response['status'], response['message'])
+        
+        data = await response.json()
+        
+    return data['data'][0]['id']
+    
+
+
+class ConnectorTwitch(Connector):
+    """A connector for Twitch."""
+    def __init__(self, config, opsdroid=None):
+        super().__init__(config, opsdroid=opsdroid)
+        _LOGGER.debug(_("Starting Twitch connector."))
+        self.name = "twitch"
+        self.opsdroid = opsdroid
+        self.listening = True
+        self.default_target = config["channel"]
+        self.token = None
+        self.code = config['code']
+        self.client_id = config['client-id']
+        self.client_secret = config['client-secret']
+        self.redirect = config.get("redirect", "http://localhost")
+        self.bot_name = config.get("bot-name", "opsdroid")
+        self.websocket = None
+        self.user_id = None
+        # TODO: Allow usage of SSL connection
+        self.server = "ws://irc-ws.chat.twitch.tv"
+        self.port = "80"
+        self.forward_url = config['forward-url']
+        # Setup routes for webhooks subscription
+        self.opsdroid.web_server.web_app.router.add_get(
+            f"/connector/{self.name}", self.handle_challenge
+        )
+        self.opsdroid.web_server.web_app.router.add_post(
+            f"/connector/{self.name}", self.twitch_webhook_handler
+        )
```

Comment:
These should be in the `connect` method. The `__init__` shouldn't do anything other than setting attributes of the instance.